### ISSUE TITLE
Add context manager for setting the current adapter

### ISSUE THREAD

**Issue by crowsonkb** [huggingface/peft#703]
### Feature request

Similar to the `model.disable_adapter()` context method, I would like a context manager for setting the active adapter (like `model.set_adapter()` but a context manager). This can also do the same thing as `model.disable_adapter()` when `None` is provided as an argument.

### Motivation

I am using several LoRAs at once during each training step (using one to evaluate/as a reward model, training another, getting logits for the KL penalty from a third) and it is hard to make sure the correct one is set at all times. I am using a stopgap right now to manage this:

```python
@contextmanager
def set_adapter(model, adapter_name):
    old_adapter_name = model.active_adapter
    try:
        if adapter_name is not None:
            model.set_adapter(adapter_name)
            yield model
        else:
            with model.disable_adapter():
                yield model
    finally:
        model.set_adapter(old_adapter_name)
```

but it should really be a feature of the peft library itself.

### Your contribution

I can test implementations of this feature.
